cognite/neat/rules/models/mapping.py

```diff
@@ -0,0 +1,67 @@
+from collections.abc import Iterator, MutableSequence, Sequence
+from typing import Any, Generic, SupportsIndex, TypeVar, get_args, overload
+
+import pandas as pd
+from pydantic import BaseModel, GetCoreSchemaHandler
+from pydantic_core import core_schema
+
+from cognite.neat.rules.models._base_rules import ClassRef, PropertyRef
+
+T_Mapping = TypeVar("T_Mapping", bound=ClassRef | PropertyRef)
+
+
+class Mapping(BaseModel, Generic[T_Mapping]):
+    source: T_Mapping
+    destination: T_Mapping
+
+
+class MappingList(list, MutableSequence[Mapping[T_Mapping]]):
```

Comment:
I went for generic here. A bit more complex, but more reusable. 
Advantage: Easier to use (nice notebook output, useful helper method `as_destination_by_source`)
Con: Harder to instantiate.

I think we just make a bunch of factory method that does the instantiation easier for us.